### PR TITLE
Update docs to use `scippnexus.v2`

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Major changes:

- The application-definition mechanism works differently now.
- We have currently removed the nx-class based properties allowing for access like `f.entry.instrument`.

### CODE REVIEW

File: docs/getting-started/quick-start-guide.ipynb [hunk 1]
```diff
@@ -64,7 +64,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import scippnexus as snx\n",
+    "# To use the legacy interface, use:\n",
+    "# import scippnexus as snx\n",
+    "import scippnexus.v2 as snx\n",
```
Comment (jl-wynen) [inline docs/getting-started/quick-start-guide.ipynb:69]: Can you explain this in a central, easy to find place? (E.g. a box on the landing page)

File: docs/user-guide/application-definitions.ipynb [hunk 2]
```diff
@@ -34,6 +34,22 @@
     "</div>"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "7fc1d425-4a31-4b75-87d4-f0398b88c090",
+   "metadata": {},
+   "source": [
+    "Definitions provide customization points, e.g., for how ScippNexus can find required information in the HDF5 group, and how contents are mapped to aspects of the returned data (typically a `scipp.DataArray` or `scipp.DataGroup`).\n",
```
Comment (jl-wynen) [inline docs/user-guide/application-definitions.ipynb:42]: Can you link to the NeXus docs?

File: docs/getting-started/quick-start-guide.ipynb (hunk 1 above)
Comment (SimonHeybrock) [reply docs/getting-started/quick-start-guide.ipynb:69]: Is it worth it, given that we will change this soon?

Comment (jl-wynen) [reply docs/getting-started/quick-start-guide.ipynb:69]: How soon? Do you want people to use v2 before then?

File: docs/user-guide/application-definitions.ipynb (hunk 2 above)
Comment (SimonHeybrock) [reply docs/user-guide/application-definitions.ipynb:42]: You mean different from the link a couple of lines above?

File: docs/getting-started/quick-start-guide.ipynb (hunk 1 above)
Comment (SimonHeybrock) [reply docs/getting-started/quick-start-guide.ipynb:69]: I think the number of people using ScippNexus directly is O(1). I think we are going to make it the default within 1-2 months?

Comment (jl-wynen) [reply docs/getting-started/quick-start-guide.ipynb:69]: Ok

File: docs/user-guide/application-definitions.ipynb (hunk 2 above)
Comment (jl-wynen) [reply docs/user-guide/application-definitions.ipynb:42]: No, this is fine. Didn't notice that the rich diff had omitted this

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**